### ISSUE TITLE
JS Errors not showing up (in the console)

### ISSUE THREAD

**Issue by Deltatiger** [angular/angular#15791]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When processing data I ran into error originating from accessing a variable with undefined as its value. However this error does not make it to the console. The rendering of the view stops and nothing happens. There is nothing on the console. 
Debugging the issue with the Developer Tools in chrome I could see that the error is thrown and it goes to core.es5.js in @angular/core/@angular/core.es5.js => callWithDebugContext() @ Line number 13054.
Whenever the errors are not propagating to the console I can debug that the state of the view is 2 in the variable _currentView.

Note:
The project was initially created from the Angular Seed Project which was running Angular 2.4.x. Recently I ported the entire application to Angular CLI 1.0.0 running Angular 4.0.0. Earlier the same issue happened where there was no issue in the console but the page was not loading (in Angular 4 / Angular-CLI 1). When investigating it was some old code present in the previous version running on Angular 2.4.x that was causing the issue and I found this out when trying to run the application on the previous versions and that caused errors on the console which I fixed in the new version (Running Angular 4)

**Expected behavior**
For any such TypeErrors the message has to propagate to the console ( when no user written catch block is specified ).

**Minimal reproduction of the problem with instructions**
I am finding it hard to reproduce in a new project since my current project is fairly large.
https://plnkr.co/edit/SbExopvcG02v74Ess5FM?p=catalogue

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

Windows 7 OS 64Bit.
Visual Studio Code 1.1.0.2
Node Package Manager 3.10.8
Using ng serve

* **Angular version:** 2.0.X

+-- @angular/animations@4.0.1
+-- @angular/cli@1.0.0
+-- @angular/common@4.0.0
+-- @angular/compiler@4.0.0
+-- @angular/compiler-cli@4.0.0
+-- @angular/core@4.0.0
+-- @angular/forms@4.0.0
+-- @angular/http@4.0.0
+-- @angular/platform-browser@4.0.0
+-- @angular/platform-browser-dynamic@4.0.0
+-- @angular/router@4.0.0
+-- @types/jasmine@2.5.38
+-- @types/node@6.0.60
+-- angular2-jwt@0.2.0
+-- bootstrap@3.3.7
+-- codelyzer@2.0.0
+-- core-js@2.4.1
+-- font-awesome@4.7.0
+-- grunt@1.0.1
+-- grunt-exec@2.0.0
+-- jasmine-core@2.5.2
+-- jasmine-spec-reporter@3.2.0
+-- jquery@3.2.1
+-- karma@1.4.1
+-- karma-chrome-launcher@2.0.0
+-- karma-cli@1.0.1
+-- karma-coverage-istanbul-reporter@0.2.0
+-- karma-jasmine@1.1.0
+-- karma-jasmine-html-reporter@0.2.2
+-- ng2-toastr@4.0.1
+-- protractor@5.1.0
+-- rxjs@5.1.0
+-- ts-node@2.0.0
+-- tslint@4.5.0
+-- typescript@2.2.0
`-- zone.js@0.8.4

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Chrome 56.0.2924.87
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
TypeScript 2.2.1

* **Node (for AoT issues):** `node --version` =   
v6.9.1

I do not know what all I need to debug this and will be glad to provide more information ( once the kind of information required is specified ).

**Comment by DzmitryShylovich** [comment]
I think this is a duplicate of https://github.com/angular/zone.js/issues/706

Can you try the fix from https://github.com/angular/zone.js/pull/709 ?

**Comment by Deltatiger** [comment]
@DzmitryShylovich Yes this seems to be making the errors popping up to the console.
For now I will patch my current zone.js from the dist folder.
Thank you.

**Comment by bigjetplane** [comment]
@Deltatiger nice. close this issue please and subscribe angular/zone.js#706 for further updates. we can't fix it in this repo anyway. thanks

**Comment by Deltatiger** [comment]
Just posting some more information before closing the issue.

The errors are properly propagating on version 0.8.0. It breaks on anything above 0.8.0. Seems Angular CLI ships with version 0.8.3 and hence the issue. 

Current solution: Dropping to version 0.8.0.

**Comment by shahjay748** [comment]
This is a really an unexpected critical bug from such a popular library. I spent around 4 hours to trace the issue which stopped everything in my app and there was no console errors. 

This issue is fixed in latest version 0.8.11.

Btw, was there any reason for not displaying errors in console ?
I just want know because I want to make sure that any future update won't do that again.

**Comment by rgarnaik** [comment]
Hello, I am now facing the same issues with my anguar app. I am using  "zone.js": "^0.8.20" version with  "@angular/core": "^5.2.0".

Not sure why suddenly this issue came up while I was upgrading some of internal component versions.  any solution?
